bundles/framework/announcements/view/AnnouncementsCollapse.jsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from 'prop-types';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import { Message, Collapse, CollapsePanel } from 'oskari-ui';
+import { AnnouncementsModal } from './AnnouncementsModal';
+import { Divider } from 'antd';
+
+//Collapse panel -> set title, content and date range according to announcement
+
+const { Panel } = Collapse;
+
+const AnnouncementsCollapse = ({controller, updated, panels, modals, checked }) => {
+
+      if (!updated) {
+        controller.getAnnouncements(function(ann){
+          console.log("get announcements collapse");
+          console.log(ann);
+
+        
+          var panels = [];
+          var modals = [];
+            ann.data.forEach((announcement) => {
+
+              if (announcement.active && controller.showModal(announcement.id)) {
+                //if announcement is active, then show pop-up of the content
+                modals.push(announcement);
+              }
+              panels.push(
+                      <CollapsePanel header={announcement.title} key={announcement.id}>
+                          <h3>{announcement.title}</h3>
+                          <p>{announcement.content}</p>
+                          <Divider />
+                          <b><Message messageKey={'valid'} /></b>
+                          <p>{announcement.begin_date} - {announcement.end_date}</p>
+                      </CollapsePanel>
+                );
+            });
+            controller.updatePanelsModals(panels,modals);
+          });
+      }
+      
+      
+  return (
+    <div>
+      {modals.map((modal, index) => {
```

Comment:
```suggestion
      {announcements.filter(a => a.active).map((modal, index) => {
```

Reply:
Aren't modals just announcements with `active:true`? Is there some reason they should be in two different arrays instead of using the same one for both?

Reply:
Modals array is an array of announcements that have been set active (to show as pop-up) AND have not been checked as 'don't show' by user ie. they are not in the localStorage array.

Reply:
This is outdated but to clarify, modals array is array of announcements that have been clicked "active" and have not been added to localStorage array.